packaging/globals.js

```diff
@@ -0,0 +1,118 @@
+var Writer = require('broccoli-writer'),
+    fs = require('fs'),
+    path = require('path'),
+    Promise = require('RSVP').Promise,
+    walk = require('walk-sync');
+
+
+var Globals = function ( inputTree, options ) {
+  if (!options) {
+    options = {};
+  }
+  if ( !( this instanceof Globals ) ) {
+    return new Globals( inputTree, options );
+  }
+  this.inputTree = inputTree;
+  this.outputPrefix = 'app';
+  // Generates global objects for files in these folders
+  this.topLevels = options.topLevels || [
+    'views',
+    'components',
+    'utils',
+    'mixins',
+    'controllers'
+  ];
+};
+
+Globals.prototype = Object.create( Writer.prototype );
+Globals.prototype.constructor = Globals;
+
+Globals.prototype.write = function (readTree, destDir) {
+  var self = this;
```

Comment:
I take it this is because we've copied the globals packaing code from elsewhere - usually we should use `_this`. Add a comment!